Fix a bug on minimum voting mana requirement

sydneyitguy opened the pull request:
![image](https://user-images.githubusercontent.com/1332279/46518892-65f32400-c8b1-11e8-8bd9-dd846e67c96c.png)

A new user will have 0 SP and 0 VP, but votings should go through even though it has zero effect on the rewards.

Currently the code change was missing, so new users are getting
`{“error”:“server_error”,“error_description”:“voter.voting_manabar.current_mana > 0: Account does not have enough mana to vote.“}` when they vote.

Temporary work-around: Delegate 0.001 SP to the new account

iamsmooth commented:
Is it even possible for voting mana to be negative? Perhaps a simpler change is to remove the test altogether, which seems more consistent with the current rules about 'dust votes'. Agree this needs to be a hard fork.

mvandeberg commented:
Voting mana cannot be negative be the amount consumed is a percentage of the remaining mana. In fact, I think it is impossible for voting mana to actually go to 0 if the user non-zero SP. This change is needed for the corner case of an account with SP, but does need to be gated by a hardfork check.

youkaicountry commented:
@sydneyitguy Please make the requested changes from code review, otherwise this PR will be closed in two weeks. We're trying to tidy up the open PRs, thanks for your understanding!

sydneyitguy commented:
@youkaicountry 

I'm not a blockchain developer, so I don't know how I can do "hardfork check".
However, I think this is a critical issue because new users cannot even vote once at the moment, so someone has to fix.. 

Can anyone help? (or should I re-create this as an issue ticket?)

mvandeberg commented:
A hardfork check is:

``` c++
if( db.has_hardfork( STEEM_HARDFORK_0_21 ) )
{
   // new logic
}
else
{
   // old logic
}
```

If you make such a change without this check then an upgraded node might have different acceptance criteria for an operation or might even have a different result in a calculation. In either case, this leads to a blockchain fork between new nodes and old nodes.

You can see this throughout the codebase. Ideally, this would also be filed as a bug report and the hardfork constant would include the issue number as well.

An example:

https://github.com/steemit/steem/blob/3730a839d88ead9e28da1642359f7054c479496c/libraries/chain/steem_evaluator.cpp#L1417-L1429

This contains all previous iterations of logic surrounding the STEEM_VOTE_DUST_THRESHOLD over several hardforks (13, 14, and 20). You can go back to the issues (#248, #259, and #1764) to see the motivation behind each change.

sydneyitguy commented:
@mvandeberg Thank you for your kind explanation 🙂
I have added a hard-fork check, and just removed the assertion on HF21 as you pointed out (voting mana cannot be negative value anyway).

mvandeberg commented:
Related to #3004

mvandeberg commented:
Fixed in #3004
